src/main.rs

```diff
@@ -1,3 +1,92 @@
+extern crate clap;
+
+use clap::{Arg, App, SubCommand};
+
 fn main() {
-    println!("Hello, world!");
+    let gpio_cmd_matches = App::new("GPIO Utils")
+        .version(env!("CARGO_PKG_VERSION"))
+        .about("Read, Write, and Configure GPIOs")
+
+        // Global options
+        .arg(Arg::with_name("config")
```

Comment:
I think some items (names/aliases) will be additive.  Others will require a resolution (with user specified config having highest priority).  Probably would be:

- `/etc/gpio.toml`
- `/etc/gpio.d/*.toml` (order would be deterministic similar to init scripts)
- User configs specified on command line (based on order specified on command line)